lib/code-ribbon-ribbon-container.js

```diff
@@ -33,11 +33,18 @@ class CodeRibbonRibbonContainer extends PaneAxis {
     // We always stack ribbons vertically
     this.orientation = "vertical";
 
+    if (state.patchQuota) {
+      this.patchQuota = state.patchQuota;
```

Comment:
for my own clarity, how is the state being saved in this code, and how is patchQuota in it

Reply:
We're extending the Atom Pane so there's some hidden functions in the Atom source code that help with the serialization for this.